src/sentry/tsdb/redis.py

```diff
@@ -381,6 +433,72 @@ def merge_aggregates(values):
             ]
         )
 
+    def merge_distinct_counts(self, model, destination, sources, timestamp=None):
+        rollups = {}
+        for rollup, samples in self.rollups.items():
+            _, series = self.get_optimal_rollup_series(
+                to_datetime(self.get_earliest_timestamp(rollup, timestamp=timestamp)),
+                end=None,
+                rollup=rollup,
+            )
+            rollups[rollup] = map(to_datetime, series)
+
+        temporary_id = uuid.uuid1().hex
+
+        def make_temporary_key(key):
+            return '{}{}:{}'.format(self.prefix, temporary_id, key)
+
+        data = {}
+        for rollup, series in rollups.items():
+            data[rollup] = {timestamp: [] for timestamp in series}
+
+        with self.cluster.fanout() as client:
+            for source in sources:
+                c = client.target_key(source)
+                for rollup, series in data.items():
+                    for timestamp, results in series.items():
+                        key = self.make_key(
+                            model,
+                            rollup,
+                            to_timestamp(timestamp),
+                            source,
+                        )
+                        results.append(c.get(key))
+                        c.delete(key)
```

Comment:
The merge task can be retried, and I'd feel better about undercounting here than overcounting.

If we're really that concerned about it, we can go through some elaborate hacks to try and make it recoverable if it errors. Personally, I don't think the complexity or effort is worth it when in the future we should just be fixing this to make it possible for the operation to be atomic anyway.